critiquebrainz/db/review.py

```diff
@@ -0,0 +1,577 @@
+               latest_revision.timestamp as latest_revision_timestamp,
+               latest_revision.text as text,
+               license.full_name,
+               license.info_url
+          FROM review
+          JOIN revision ON review.id = revision.review_id
+     LEFT JOIN vote ON vote.revision_id = revision.id
+          JOIN "user" ON review.user_id = "user".id
+          JOIN license ON license.id = license_id
+          JOIN (
+                revision
+                JOIN (
+                    SELECT review.id AS review_uuid,
+                           MAX(timestamp) AS latest_timestamp
+                      FROM review
+                      JOIN revision ON review.id = review_id
+                  GROUP BY review.id
+                  ) AS latest
+                  ON latest.review_uuid = revision.review_id
+                 AND latest.latest_timestamp = revision.timestamp
+               ) AS latest_revision ON review.id = latest_revision.review_id
+        {where_clause}
+      GROUP BY review.id, latest_revision.id, "user".id, license.id
+        {order_by_clause}
+         LIMIT :limit
+        OFFSET :offset
+        """.format(where_clause=filterstr, order_by_clause=order_by_clause))
+
+    filter_data["limit"] = limit
+    filter_data["offset"] = offset
+
+    with db.engine.connect() as connection:
+        results = connection.execute(query, filter_data)
+        rows = results.fetchall()
+        rows = [dict(row) for row in rows]
+        # Orgainze last revision info in reviews
+        if rows:
+            for row in rows:
+                row["last_revision"] = {
+                    "id": row.pop("latest_revision_id"),
+                    "timestamp": row.pop("latest_revision_timestamp"),
+                    "text": row["text"],
+                    "review_id": row["id"],
+                }
+                row["user"] = User({
+                    "id": row["user_id"],
+                    "display_name": row.pop("display_name"),
+                    "show_gravatar": row.pop("show_gravatar"),
+                    "is_blocked": row.pop("is_blocked"),
+                    "musicbrainz_username": row.pop("musicbrainz_id"),
+                    "email": row.pop("email"),
+                    "created": row.pop("user_created"),
+                })
+    return rows, count
+
+
+def get_popular(limit=None):
+    """Get list of popular reviews.
+
+    Popularity is determined by rating of a particular review. Rating is a difference between
+    positive votes and negative. In this case only votes from the last month are
+    used to calculate rating.
+
+    Results are cached for 1 hour.
+
+    Args:
+        limit: Maximum number of reviews to return.
+
+    Returns:
+        Randomized list of popular reviews which are converted into dictionaries using to_dict method.
+    """
+    cache_key = cache.gen_key("popular_reviews", limit)
+    reviews = cache.get(cache_key, REVIEW_CACHE_NAMESPACE)
+    reviews = None
```

Comment:
What's the point of retrieving reviews from cache and then getting rid of them in the next statement?